CookingCurator/Controllers/Manager.cs

```diff
@@ -95,28 +99,70 @@ public RecipeBaseViewModel RecipeAdd(RecipeAddViewModel recipe)
             return addedItem == null ? null : mapper.Map<RECIPE, RecipeBaseViewModel>(addedItem);
         }
 
-        public RecipeBaseViewModel RecipeEdit(int id, RecipeAddViewModel recipe)
+        public RecipeBaseViewModel RecipeEdit(Recipe_IngredViewModel recipeIng)
         {
-            var recipeUpdate = ds.Recipes.Find(id);
+            var recipeUpdate = ds.Recipes.Find(recipeIng.recipe.recipe_Id);
             if (recipeUpdate == null)
             {
                 return null;
             }
-            recipeUpdate.title = recipe.title;
-            recipeUpdate.instructions = recipe.instructions;
+            recipeUpdate.title = recipeIng.recipe.title;
+            recipeUpdate.instructions = recipeIng.recipe.instructions;
             recipeUpdate.lastUpdated = DateTime.Now;
-            recipeUpdate.author = recipe.author;
-            recipeUpdate.source_Link = recipe.source_Link;
-            recipeUpdate.country = recipe.country;
-            recipeUpdate.mealTimeType = recipe.mealTimeType;
-            ds.Entry(recipeUpdate).State = EntityState.Modified;
+            recipeUpdate.author = recipeIng.recipe.author;
+            recipeUpdate.source_Link = recipeIng.recipe.source_Link;
+            recipeUpdate.country = recipeIng.recipe.country;
+            recipeUpdate.mealTimeType = recipeIng.recipe.mealTimeType;
+            ds.Entry(recipeUpdate).State = System.Data.Entity.EntityState.Modified;
+
+            deleteIngredients(recipeIng.recipe.recipe_Id);
+            addIngredients(recipeIng.recipe.recipe_Id, recipeIng.selectedIngredsId);
             // Attempt to save the edited recipe.
             ds.SaveChanges();
 
             // If successful, return the edited recipe (mapped to a view model class).
             return recipeUpdate == null ? null : mapper.Map<RECIPE, RecipeBaseViewModel>(recipeUpdate);
         }
 
+        public void RecipeDelete(int id)
+        {
+            deleteIngredients(id);
+            var recipe = ds.Recipes.Find(id);
+            ds.Recipes.Remove(recipe);
+            ds.SaveChanges();
+        }
+
+        public IEnumerable<IngredientBaseViewModel> IngredientGetAll()
+        {
+            return mapper.Map<IEnumerable<INGRED>, IEnumerable<IngredientBaseViewModel>>(ds.Ingreds);
+        }
+
+        public void addIngredients(int id, String[] selectedIds)
```

Comment:
Could we rename this method? It might be confusing if we have a INGRED add method later on.